code/modules/mob/living/simple_animal/hostile/carp.dm

```diff
@@ -83,10 +85,22 @@
 		else
 			make_tameable()
 
+/mob/living/simple_animal/hostile/carp/revive(full_heal, admin_revive)
+	if (tamed)
+		var/datum/weakref/friendref = ai_controller.blackboard[BB_HOSTILE_FRIEND]
+		tamed(friendref.resolve())
```

Comment:
```suggestion
		var/datum/weakref/friendref = ai_controller.blackboard[BB_HOSTILE_FRIEND]
		var/mob/living/friend = friendref?.resolve()
		if(friend)
			tamed(friend)
```
it doesn't seem guaranteed that friendref will exist, or that even friendref's ref will exist.